### ISSUE TITLE
Assert: Closing the Material Editor prints an assert when launched via Visual Studio 2019

### ISSUE THREAD

**Issue by LB-ArturZieba** [o3de/o3de#8259]
**Describe the bug**
Closing the Material Editor prints an assert to the Microsoft Visual Studio Debug Console and the Visual Studio 2019 Debug Output Log. Please refer to the logs, and the screenshot attached in the **Logs, screenshot, and custom.txt** section for more information.

Assert information:
File:
```
C:\o3de\Code\Framework\AzCore\AzCore/EBus/Internal/Handlers.h
```
Condition:
```
(1088) '__cdecl AZ::Internal::NonIdHandler<class AzToolsFramework::AssetSystem::AssetSystemNotifications,class AzToolsFramework::AssetSystem::AssetSystemNotifications,struct AZ::Internal::EBusContainer<class AzToolsFramework::AssetSystem::AssetSystemNotifications,class AzToolsFramework::AssetSystem::AssetSystemNotifications,0,1> >::~NonIdHandler(void)'
EBus handlers must be disconnected prior to destruction on multi-threaded buses with virtual functions
```
Line: 187

**Steps to reproduce**
1. Make sure that all O3DE applications are closed (Editor, Asset Processor, Material Editor, Lua Editor etc.).
2. Paste the custom.txt file attached in the **Logs, screenshot, and custom.txt** section to your ...\.o3de\Registry\ folder after adjusting its "project_path" parameter to match the existing project (Example path: `C:\Users\GPC-610\.o3de\Registry\custom.setreg`), and changing its extension from .txt to .setreg (custom.setreg).
3. Launch the AssetProcessor.exe from your build path (Example path: `C:\o3de-install\bin\Windows\profile\Default\AssetProcessor.exe`).
4. Launch Visual Studio 2019.
5. Go to File > Open > Project/Solution... and choose your Material Editor executable (Example path: `C:\o3de-install\bin\Windows\profile\Default\MaterialEditor.exe`).
6. Start the loaded Material Editor via the Visual Studio 2019 (Click the "Green play button" Start on the top toolbar).
7. Wait for the Material Editor to fully load, a "Microsoft Visual Studio Debug Console" should also launch alongside with it.
8. Verify that the Material Editor is functional by opening the material mentioned in the **Assets required** section.
9. Close the Material Editor by clicking its window Close button (X in the top-right corner).

**Expected behavior**
Material Editor closes with no issues.

**Actual behavior**
As the Material Editor is closing, an assert is printed to the Microsoft Visual Studio Debug Console and the Visual Studio 2019 Debug Output Log.

**Assets required**
* Visual Studio 2019 installed.
* A project is built (All Gems enabled, and Default Template projects were used for reproducing this issue).
* A material is made, and saved via the Material Editor.

**Logs, screenshot, and custom.txt**
[MSVSDebugConsoleLog.txt](https://github.com/o3de/o3de/files/8232239/MSVSDebugConsoleLog.txt)
[VS2019DebugOutputLog.txt](https://github.com/o3de/o3de/files/8232240/VS2019DebugOutputLog.txt)
![VSAssert](https://user-images.githubusercontent.com/86952082/157880484-dfa617f9-c28f-4a3f-9b45-08c33837cd99.PNG)
[custom.txt](https://github.com/o3de/o3de/files/8232238/custom.txt)

**Found in Branch**
Development (487c454)

**Desktop**
 - Device: PC
 - OS: Windows
 - Version 10
 - CPU AMD Ryzen 5 3600
 - GPU NVIDIA GeForce RTX 2060 SUPER
 - Memory 16GB

**Comment by monroegm** [comment]
Rerouting to SIG-Graphics-Audio because this has to do with the Material Editor

**Comment by LB-ArturZieba** [comment]
Issue verified as fixed on the Development (6896fcb) build.
